cubedash/_pages.py

```diff
@@ -273,10 +274,34 @@ def _get_grouped_products() -> List[Tuple[str, List[ProductWithSummary]]]:
     # Which field should we use when grouping products in the top menu?
     group_by_field = app.config.get("CUBEDASH_PRODUCT_GROUP_BY_FIELD", "product_type")
     group_field_size = app.config.get("CUBEDASH_PRODUCT_GROUP_SIZE", 5)
+    group_by_regex = app.config.get("CUBEDASH_PRODUCT_GROUP_BY_REGEX", None)
 
-    # Group using the configured key, or fall back to the product name.
-    def key(t):
-        return t[0].fields.get(group_by_field) or t[0].name
+    if group_by_regex:
+        try:
+            regex_group = {}
+            for regex, group in group_by_regex:
+                regex_group[re.compile(regex.strip())] = group.strip()
+        except Exception:
+            _LOG.warn(
+                "invalid CUBEDASH_PRODUCT_GROUP_BY_REGEX: {}".format(group_by_regex)
+            )
+            group_by_regex = None
```

Comment:
I think we should just throw a hard error. If they specify a config option it should be valid.

Something like:
```suggestion
        except re.error as e:
            raise RuntimeError(
                f"Invalid regexp in CUBEDASH_PRODUCT_GROUP_BY_REGEX {group!r}: {e!r}"
            )
```
(also catching the specific exception, and giving user the location and message)